### ISSUE TITLE
Communicate failed checks back to the author/project

### ISSUE THREAD

**Issue by djdelorie** [getpatchwork/patchwork#425]
In our project, we've had instances of developers being surprised that they had failed checks - or that checks were even happening.  Rather than force developers to check a web page when they're used to using email, it would be nice if the patchwork server could send out some form of status when checks fail (or perhaps also when they succeed); either to the patch author or to the  project mailing list.

Yes, I know a separate app could poll the patchwork API and send emails as needed, but doing it in patchwork results in more immediate feedback and less processing.  Reporting from whatever app is doing the checks would require all check-generators to support email independently, and distribute any configuration requirements.

**Comment by chantra** [comment]
We are thinking about something similar for our project too..

Agreed that it could be done by apps themselves, but that would be a lot of re-implementation going on when PW seems to be in a good position to provide this to a wider set of projects.

I suppose a given project does not necessarily want to receive an email for all checks that may be ran, some may want to receive update for success, warning, fail. Some may only want to receive fail state updates...

Some may want to send email to maintainers, to mailing list thread, or to patch author.

So, those would probably be options that would need to be made available at the project level.

**Comment by chantra** [comment]
@stephenfin , is there any interest in this feature? Would you be interested in a PR providing it?

**Comment by stephenfin** [comment]
Absolutely. I just don't have the time to do it myself but I'd happily review a PR.

**Comment by kuba-moo** [comment]
Please make sure there is a per-project opt-out, tho! In Linux networking we try to shame people for posting stuff that clearly doesn't work, because otherwise we get an insane number of patch postings just to run the CI :(  Bot replies are too impersonal for us, but I'm sure smaller patchwork.kernel.org projects would like to enable this.
